web/lib/firebase/comments.ts

```diff
@@ -62,22 +56,59 @@ export async function createCommentOnGroup(
   replyToCommentId?: string
 ) {
   const ref = doc(getCommentsOnGroupCollection(groupId))
+  return await createComment(
+    groupId,
+    'group',
+    content,
+    user,
+    ref,
+    replyToCommentId
+  )
+}
+
+export async function createCommentOnPost(
+  postId: string,
+  content: JSONContent,
+  user: User,
+  replyToCommentId?: string
+) {
+  const ref = doc(getCommentsOnPostCollection(postId))
+
+  return await createComment(
+    postId,
+    'post',
+    content,
+    user,
+    ref,
+    replyToCommentId
+  )
+}
+
+async function createComment(
+  surfaceId: string,
+  surfaceType: 'contract' | 'group' | 'post',
+  content: JSONContent,
+  user: User,
+  ref: DocumentReference<DocumentData>,
+  replyToCommentId?: string,
+  extraFields: { [key: string]: any } = {}
+) {
   const comment = removeUndefinedProps({
     id: ref.id,
-    commentType: 'group',
-    groupId,
     userId: user.id,
     content: content,
     createdTime: Date.now(),
     userName: user.name,
     userUsername: user.username,
     userAvatarUrl: user.avatarUrl,
     replyToCommentId: replyToCommentId,
+    ...extraFields,
   })
-  track('group message', {
+
+  track(`${surfaceType} message`, {
     user,
     commentId: ref.id,
-    groupId,
+    surfaceId,
```

Comment:
What do you mean by "for backwards compatibility of the events"?

Reply:
`track()` creates an event in Amplitude with these props.

We could look at stats that compare contract comments vs group comments vs post comments, but with this change, new events would be harder to compare with events before this change with `surfaceId` instead of `contractId` etc. It's not a huge deal, but it seems like it would be good to include the same props as before (adding new props is fine).

Reply:
Ahh, I see. I'll fix this - already merged this one but I'll put it in a separate PR.